storage/src/main/java/org/apache/kafka/server/log/remote/metadata/storage/TopicBasedRemoteLogMetadataManager.java

```diff
@@ -327,6 +327,17 @@ public void onStopPartitions(Set<TopicIdPartition> partitions) {
         }
     }
 
+    @Override
+    public Long remoteLogSize(TopicIdPartition topicPartition, int leaderEpoch) throws RemoteStorageException {
+        long remoteLogSize = 0L;
+        Iterator<RemoteLogSegmentMetadata> remoteLogSegmentMetadataIterator = remotePartitionMetadataStore.listRemoteLogSegments(topicPartition, leaderEpoch);
```

Comment:
Please add a comment here explaining that this is a naive implementation and if required, could be improved in future. I know that you mentioned it in the description of the ticket but the code reader perhaps should also know why a decision was made and what can be done here.

Reply:
Hopefully the comment in the newest commit covers all of these. If not let me know and I will amend it!

Reply:
I didn't see the comment you added.

Reply:
Is this remedied now for you @showuon? The comment is
```
// This is a simple-to-understand but not the most optimal solution.
// The TopicBasedRemoteLogMetadataManager's remote metadata store is file-based. During design discussions
// at https://lists.apache.org/thread/kxd6fffq02thbpd0p5y4mfbs062g7jr6
// we reached a consensus that sequential iteration over files on the local file system is performant enough.
// Should this stop being the case, the remote log size could be calculated by incrementing/decrementing
// counters during API calls for a more performant implementation.
```